pyscf/fci/addons.py

```diff
@@ -658,22 +658,36 @@ def transform_ci(ci, nelec, u):
     nb_new = cistring.num_strings(norb_new, nelecb)
     ci = ci.reshape(na_old, nb_old)
 
-    one_particle_strs_old = numpy.asarray([1 << i for i in range(norb_old)])
-    one_particle_strs_new = numpy.asarray([1 << i for i in range(norb_new)])
+    one_particle_strs_old = numpy.asarray(cistring.make_strings(range(norb_old), 1))
+    one_particle_strs_new = numpy.asarray(cistring.make_strings(range(norb_new), 1))
 
     if neleca == 0:
         trans_ci_a = numpy.ones((1, 1))
     else:
         trans_ci_a = numpy.zeros((na_old, na_new), dtype=ua.dtype)
-        strs_old = numpy.asarray(cistring.make_strings(range(norb_old), neleca))
 
         # Unitary transformation array trans_ci is the overlap between two sets of CI basis.
-        occ_masks_old = (strs_old[:,None] & one_particle_strs_old) != 0
+        strs_old = numpy.asarray(cistring.make_strings(range(norb_old), neleca))
+        if norb_old <= 64:
```

Comment:
* Should be `norb_old < 64`. make_strings function creates the other type of string representation for 64 orbitals
* Consider to put the occ_mask initialization code in a separated function. They are repeated many times in this function.

Reply:
@sunqm Thanks. Your comments have been implemented.